reactive/INTRODUCTION_EN.md

```diff
@@ -0,0 +1,153 @@
+I like exploding things, so we'll talk about a cannon.
+Don't worry about the `(x)` right now, you will understand them later on…
+
+Let's suppose you happily go to Wallmart to buy a cannon. It's a high-tech cannon that can automagically recharge.
+With your newfound super-duper cannon, you go back and `(1)` install it on a hill (locking on your boss), `(2)` gets the fuses batch, and `(3)` fire as many fuses as you want to throw cannonballs on your boss.
+- You don't know when the cannon will fire.
+- You don't even know whether the cannon will throw enough cannonballs to match your explosion desires (Has it enough ammunition? Will it explode halfway?)
+- You hate your boss enough that you decided to dance whenever the cannon throws a cannonball.
+- You somehow decided to run if the cannon explodes. Thanks, Captain Obvious.
+- You have decided to go on strike if your boss is still dead and you are running out of cannonballs. And if the cannon hasn't exploded, otherwise you would have fled too far~
+
+That's a nice program we got here!
+
+> Shouldn't this sheet actually be concerned about programming? We only discussed cannoning so far…
+
+Oh my, how silly of me. I don't want to change topics so let's implement this high-tech cannon in pseudo java code!
+
+```java
+// This cannon is so high tech that it may or may not possess an infinite amount of cannonballs.
+interface Cannon {
+	// To install the cannon on our lovely hill, and take the time to accurately target your boss.
+	void install(Me me);
+}
+
+// I didn't declare a lot of things, we are doing pseudocode here
+interface Me {
+	// When the cannon finished it's installation, it will notify us and we get an infinite batch of fuses.
+  // Just because it's the future.
+	void onPrepare(FuseBatch fuses);
+	
+	// This is called whenever the cannon launches a cannonball toward your patron.
+	default void onNext(CannonBall ball) {
+		dance();
+	}
+
+	// This is called when the cannon of the future throws an uncaught runtime exception and explodes.
+  // Yes, the futuuuuuuuure still has runtime exceptions :'c
+	default void onError(Throwable t) {
+		runForYourLife();
+	}
+	
+	// This is called when the cannon has no cannonball left to fire.
+	default void onComplete() {
+		if (boss.isAliveAndKicking())
+			timeToGoOnStrike();
+	}
+}
+
+interface FuseBatch {
+	// We kindly ask the cannon to launch `fuseAmount` cannonballs by firing as many fuses.
+	void fire(long fuseAmount);
+	
+	// If you suddendly remember the inherent kindness of your boss, stow your cannon and try to cancel the cannonballs' throwing.
+  // You might miss some fuses though, trampling on them isn't exactly a precise method.
+	void cancel();
+}
+```
+
+Well now that's done, we don't have anything else to do with our cannon.
+So let's haphazardly pick some code from [reactive-streams](http://www.reactive-streams.org/) since it should be related to the official topic of this sheet.
```

Comment:
```suggestion
So let's haphazardly pick some code from [reactive-streams](http://www.reactive-streams.org/) since it should be related to the official topic of this article.
```